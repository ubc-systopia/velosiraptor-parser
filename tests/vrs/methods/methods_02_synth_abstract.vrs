segment MyUnit {


    // abstract method
    abstract fn foo() -> bool
        requires true;

    abstract fn foo() -> bool
        requires true;
    {
        true
    }


    // synthesis method
    synth fn foo() -> bool
        requires true;


    synth fn foo() -> bool
        requires true;
    {
        true
    }


    // synth + abstract method
    abstract synth fn foo() -> bool
        requires true;

    // synth + abstract method
    abstract synth fn foo() -> bool
        requires true;
    {
        true
    }
}