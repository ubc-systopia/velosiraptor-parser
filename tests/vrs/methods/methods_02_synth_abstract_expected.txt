VelosiParseTree(tests/vrs/methods/methods_02_synth_abstract.vrs)
---------------------------------------------
segment MyUnit {
  abstract fn foo() -> bool
    requires true

  abstract fn foo() -> bool
    requires true
  {
    true
  }

  synth fn foo() -> bool
    requires true

  synth fn foo() -> bool
    requires true
  {
    true
  }

  abstract synth fn foo() -> bool
    requires true

  abstract synth fn foo() -> bool
    requires true
  {
    true
  }

}


---------------------------------------------
